<launch>

    <arg name="tf_name" default="" doc="nome del singolo robot per casi multirobot"/>

    <arg name="slash" value="$(eval '/' if tf_name!='' else '')" doc="argomento ausiliario per gestire tf frame"/>

    <node pkg="move_base" type="move_base" respawn="true" name="move_base" output="log">
        <!-- <remap from="map" to="/map"/> -->
        <param name="base_global_planner" value="global_planner/GlobalPlanner" />
        <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
        <rosparam file="$(find test_unknown_rendezvous)/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find test_unknown_rendezvous)/params/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find test_unknown_rendezvous)/params/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find test_unknown_rendezvous)/params/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find test_unknown_rendezvous)/params/move_base_params.yaml" command="load" />
        <rosparam file="$(find test_unknown_rendezvous)/params/dwa_local_planner_params.yaml" command="load" />
        <rosparam file="$(find test_unknown_rendezvous)/params/global_planner_params.yaml" command="load" />
        <param name="global_costmap/robot_base_frame" value="$(arg tf_name)$(arg slash)base_footprint"/>
        <param name="local_costmap/robot_base_frame"  value="$(arg tf_name)$(arg slash)base_footprint"/>
        <param name="global_costmap/global_frame"     value="$(arg tf_name)$(arg slash)map"/>
        <param name="local_costmap/global_frame"      value="$(arg tf_name)$(arg slash)odom"/>
        <param name="local_costmap/scan/sensor_frame" value="$(arg tf_name)$(arg slash)base_scan"/>
    </node>

</launch>